### PR TITLE
 Add basic railroad network generation and mod to enable it

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary
Features "Railroads mod"

#### Purpose of change

Add basic railroad network generation and mod to enable it.

#### Describe the solution

Added `railroads` mod, that enables railroad network generation.

#### Describe alternatives you've considered

Wait until someone implements railroads properly.

#### Testing

Create a game with `railroads` mod enabled and see that railroads are generated.

#### Additional context

Generation is pretty basic right now (it is mostly the same algorithm as regular road generation with some parameters tweaked). I expect people would play with the mod and hopefully propose more fine-tuned parameters, start updating mapgen parts of railroad, work on proper bridges, railroad engine spawn, expand on overmap special connections, etc.

image

https://github.com/CleverRaven/Cataclysm-DDA/assets/16213433/0afc1c55-d609-48bb-b68a-cf689285a237

### CODE REVIEW

File: data/mods/railroads/modinfo.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[
+  {
+    "type": "MOD_INFO",
+    "id": "railroads",
+    "name": "Railroads",
+    "authors": [ "ZhilkinSerg" ],
+    "description": "Enables basic railroad network generation in the world. <color_red>Not intended for regular gameplay, more for development purposes.</color>",
```
Comment (ZhilkinSerg) [inline data/mods/railroads/modinfo.json:7]:
```suggestion
    "description": "Enables basic railroad network generation in the world.  <color_red>Not intended for regular gameplay, more for development purposes.</color>",
```

**Comment by Procyonae** [comment]
Maybe make rail stations 0 occurrences in vanilla but enabled them in this and remove the no rail station mod too?

**Comment by ZhilkinSerg** [comment]
Don't wanna touch railroad stations to move them across existing mods, but we should definitely make use of them here evntually.

**Comment by Procyonae** [comment]
Oh I didn't realise this isn't even using the stations yet fair enough

**Comment by guo87995585** [comment]
Great work! Looking forward to iterative updates！I dream of driving armored trains in CDDA! But unfortunately, there are issues that I can't do now
Cars cannot cross the track normally (in reality, track crossing is allowed for cars to pass, and the track will be at the same height as the ground)
The double track railway is too complex, and it depends on God's arrangement where the train can travel. Why not design it as a single track railway, but this may also require solving the problem of trains crossing the line and turning
The train is unable to connect the train station to the steelmaking plant (train station), which is a bit regrettable, but it is still acceptable (clean track is the correct choice, considering gameplay, it is best not to have ready-made trains on the track)

**Comment by guo87995585** [comment]
9 G6 LI{@2JNP6D1 5N$S6

This road is not a problem for highways, but it is difficult for railways to accept

**Pull request merged**